### PR TITLE
Remove `additionalProperties` from JDBC source connectors

### LINKED ISSUE #13540: Fix integration tests source-cockroachdb Mac OS
After updating testcontainers lib to 1.17.1 version integration tests failed for Mac OS with next error:

```
java.sql.SQLTransientConnectionException: HikariPool-1 - Connection is not available, request timed out after 30007ms.
at com.zaxxer.hikari.pool.HikariPool.createTimeoutException(HikariPool.java:696)
```

### LINKED ISSUE #14549: Remove `additionalProperties: false` from JDBC source connectors
## Problem
As discovered in https://github.com/airbytehq/oncall/issues/289, a spec declaring "additionalProperties": false introduces the risk of accidental breaking changes. Specifically, when removing a property from the spec, existing connector configs will no longer be valid.

We are about to make changes that will complicate rolling back connectors that have `additionalProperties` set to `false`:

- https://github.com/airbytehq/airbyte/issues/13840
- https://github.com/airbytehq/airbyte/issues/12917
- https://github.com/airbytehq/airbyte/issues/4034

to make these upgrades/rollbacks smoother, we need to remove `additionalProperties` from connectors affected by these changes.  
## Action items

- [ ] remove `additionalProperties:false` (or replace it with `additionalProperties:true`, whichever is the correct approach)
- [ ] publish new versions of JDBC-based connectors
- [ ] merge changes
- [ ] comment on related PRs to mention that this has been done and they can proceed

### PR DESCRIPTION
**Pull request by VitaliiMaltsev**
## What
As discovered in https://github.com/airbytehq/oncall/issues/289, a spec declaring "additionalProperties": false introduces the risk of accidental breaking changes. Specifically, when removing a property from the spec, existing connector configs will no longer be valid.

To make these upgrades/rollbacks smoother, we need to remove additionalProperties from connectors affected by these changes.

## How
Remove additionalProperties: false

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Review by sashaNeshcheret — APPROVED** [review]

**Review by alexandr-shegeda — APPROVED** [review]
LGTM

File: airbyte-integrations/connectors/source-snowflake/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/SnowflakeSourceAuthAcceptanceTest.java [hunk 1]
```diff
@@ -20,7 +20,7 @@ public class SnowflakeSourceAuthAcceptanceTest extends SnowflakeSourceAcceptance
   protected DataSource createDataSource() {
     final HikariDataSource dataSource = new HikariDataSource();
     final Properties properties = new Properties();
-    final JsonNode config = getStaticConfig();
+    config = getStaticConfig();
```
Comment (grishick) [inline airbyte-integrations/connectors/source-snowflake/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/SnowflakeSourceAuthAcceptanceTest.java:23]: why is `final` removed as part of this change?

**Review by grishick — CHANGES_REQUESTED** [review]
looks like one rogue change snuck into this PR (final keyword removed) and the PR needs to be rebased. Otherwise - looks good.

**Comment by VitaliiMaltsev** [comment]
> looks like one rogue change snuck into this PR (final keyword removed) and the PR needs to be rebased. Otherwise - looks good.

@grishick no, this change is necessary. The SnowflakeSourceAuthAcceptanceTest test is broken in the master branch and redefining the global variable _config_ instead of defining a new local variable is a fix for this issue

**Review by grishick — APPROVED** [review]

**Review by tuliren — APPROVED** [review]

**Comment by VitaliiMaltsev** [comment]
/publish connector=connectors/source-clickhouse-strict-encrypt auto-bump-version=false

**Comment by VitaliiMaltsev** [comment]
/publish connector=connectors/source-cockroachdb-strict-encrypt auto-bump-version=false

**Comment by VitaliiMaltsev** [comment]
/publish connector=connectors/source-db2-strict-encrypt auto-bump-version=false

**Comment by VitaliiMaltsev** [comment]
/publish connector=connectors/source-mssql-strict-encrypt auto-bump-version=false

**Comment by VitaliiMaltsev** [comment]
/publish connector=connectors/source-mysql-strict-encrypt auto-bump-version=false

**Comment by VitaliiMaltsev** [comment]
/publish connector=connectors/source-oracle-strict-encrypt auto-bump-version=false

**Comment by VitaliiMaltsev** [comment]
/publish connector=connectors/source-clickhouse-strict-encrypt auto-bump-version=false

**Comment by VitaliiMaltsev** [comment]
/publish connector=connectors/source-clickhouse-strict-encrypt auto-bump-version=false

**Comment by grishick** [comment]
This can be merged now.

**Comment by grishick** [comment]
@VitaliiMaltsev please file separate issues for fixing the broke integration tests (the ones that are failing on this PR) and then you can merge this PR.

**Comment by VitaliiMaltsev** [comment]
@grishick  I published all connectors except _clickhouse-source_ and _clickhouse-source-strict-encrypt_ because integration tests are broken there. The publishing of these connectors will be part of [this PR](https://github.com/airbytehq/airbyte/pull/14701)

**Pull request merged**